### ISSUE TITLE
Site name is undefined unless Forgot Passsword has been configured

### ISSUE THREAD

**Issue by hjelmevold** [enonic/app-simple-idprovider#63]
If you set up a simple id provider and enable 2FA but never enable Forgot Password, the e-mail that is sent with the 2FA code will have subject "Login to undefined" and "Your code to login to undefined is:"

Summary of possible states:
1. 2FA disabled, Forgot Password disabled: site name is undefined (but to my knowledge not used by the code)
2. 2FA enabled, Forgot Password disabled (never enabled): site name cannot be configured (will be undefined)
3. 2FA enabled, Forgot Password enabled: site name is configurable
4. 2FA disabled, Forgot Password enabled: site name is configurable
5. 2FA enabled, Forgot Password disabled (but previously configured): site name will be what was previously configured

In other words, a workaround is to first enable Forgot Password, configure site name, save, then disable Forgot Password, save. Then we will be in state number 5 above.

The site name is undefined due to this code:
https://github.com/enonic/app-simple-idprovider/blob/master/src/main/resources/lib/config.js#L43-L47

The schema is here:
https://github.com/enonic/app-simple-idprovider/blob/master/src/main/resources/idprovider/idprovider.xml#L46-L59

**Comment by hjelmevold** [comment]
I guess one way to fix this would be to move the site name field out of the `forgotPassword` item-set? Alternatively, add a similar site name field to the `twoFactorEmail` item-set and fetch that separately in the code.
